libdevice/device.h

```diff
@@ -15,26 +15,18 @@
 #define EXTERN_C
 #endif // __cplusplus
 
-#ifdef CL_SYCL_LANGUAGE_VERSION
-#ifndef SYCL_EXTERNAL
-#define SYCL_EXTERNAL
-#endif // SYCL_EXTERNAL
-
+#if __SPIR__
+#define __SPIR_TARGET_ONLY__ 1
 #ifdef __SYCL_DEVICE_ONLY__
 #define DEVICE_EXTERNAL SYCL_EXTERNAL __attribute__((weak))
 #else // __SYCL_DEVICE_ONLY__
-#define DEVICE_EXTERNAL static
-#undef EXTERN_C
-#define EXTERN_C
+#define DEVICE_EXTERNAL __attribute__((weak))
 #endif // __SYCL_DEVICE_ONLY__
-#else  // CL_SYCL_LANGUAGE_VERSION
-#define DEVICE_EXTERNAL
-#endif // CL_SYCL_LANGUAGE_VERSION
 
 #define DEVICE_EXTERN_C DEVICE_EXTERNAL EXTERN_C
 
-// We need the following header to ensure the definition of all spirv variables
-// required by the wrapper libraries.
-#include "spirv_vars.hpp"
+#else // __SPIR__
+#define __SPIR_TARGET_ONLY__ 0
```

Comment:
Is this macro is needed for non-spir targets? Can it be defined only for spir?

Reply:
One more question: why can't just use `__SPIR__`?

Reply:
> Is this macro needed for non-spir targets? Can it be defined only for spir?

I want to make sure that this header file is the only place to control it, so I want to have this macro defined in all compilations.  If it is (accidentally) redefined, then the build will break making the issue evident.  This is a devicelib internal macro that I want to have full control over.

Reply:
> One more question: why can't just use `__SPIR__`?

It makes any future manipulations with the implementation guard simpler.  Not that I expect any further modifications, but I already did it once (when I removed __SYCL_DEVICE_ONLY__), and it required changes in all files.

Reply:
> If it is (accidentally) redefined, then the build will break making the issue evident.

I don't know if it is a real issue. Compiler does not define this macro, users do not define it either. Header defines it in a single place surrounded by include guards.

Macros such as this are usually not defined on "other" platforms. This way `__SYCL_DEVICE_ONLY__` is not defined for host compilation (as opposed to be defined to 0) and `__X86_64__` is not defined for arm.

Reply:
I am not worried about users.  I am worried about developers :) `__SPIR_TARGET_ONLY__` kind of invades onto FE's territory.  If at any point it is end up being defined for whatever reason, I want to be aware of it.

If you do not like `#define __SPIR_TARGET_ONLY__ 0`, I can replace it with `#undef __SPIR_TARGET_ONLY__` in the else clause, but I want to have the controlled behavior in all compilations.

Reply:
Why not choose a unique prefix, such as `__DEVICELIB_DEVICE_ONLY__`?

Reply:
There is no such thing as unique name, unless it is already defined by FE.

I think we waste too much time discussing this.  How about I get rid of `__SPIR_TARGET_ONLY__` and use just `__SPIR__` instead?  I believe everybody agrees with that, am I right?